### PR TITLE
Feature/cachebox 57 lucee provider config fix

### PR DESCRIPTION
**Pull request by sjdaniels**

### CODE REVIEW

File: system/cache/providers/CFProvider.cfc [hunk 1]
```diff
@@ -212,10 +212,13 @@ component
      * Get a structure of all the keys in the cache with their appropriate metadata structures. This is used to build the reporting.[keyX->[metadataStructure]]
      */
     struct function getStoreMetadataReport(){
-		return getKeys()
-			.map( function( item ){
-				return getCachedObjectMetadata( item );
+		var result = {};
+		getKeys()
```
Comment (lmajano) [inline system/cache/providers/CFProvider.cfc]: hmm, what is the difference of using map() vs this approach? Map does exactly the same

File: system/cache/providers/CFProvider.cfc [hunk 2]
```diff
@@ -538,14 +541,7 @@ component
...
 		 // Add in settings not discovered
-		structAppend( variables.configuration, variables.DEFAULTS );
-		// Validate configuration values, if they don't exist, then default them to DEFAULTS
-		for( var key in variables.DEFAULTS ){
-			if( NOT len( variables.configuration[ key ] ) ){
-				variables.configuration[ key ] = variables.DEFAULTS[ key ];
-			}
-		}
-
+		structAppend( variables.configuration, variables.DEFAULTS, false );
```
Comment (lmajano) [inline system/cache/providers/CFProvider.cfc]: Good point, right now it DOES override it.

File: system/cache/providers/lucee-lib/LuceeStats.cfc [hunk 3]
```diff
@@ -71,7 +71,7 @@ component implements="coldbox.system.cache.util.IStats" accessors="true"{
 	/**
 	 * Get the total cache's hits
 	 */
-	numeric function getHits();
+	numeric function getHits() {
```
Comment (lmajano) [inline system/cache/providers/lucee-lib/LuceeStats.cfc]: Good catch

File: system/cache/providers/LuceeProvider.cfc [hunk 4]
```diff
@@ -111,10 +111,13 @@ component
      * Get a structure of all the keys in the cache with their appropriate metadata structures. This is used to build the reporting.[keyX->[metadataStructure]]
      */
     struct function getStoreMetadataReport(){
-		return getKeys()
-			.map( function( item ){
-				return getCachedObjectMetadata( item );
+		var result = {};
```
Comment (lmajano) [inline system/cache/providers/LuceeProvider.cfc]: Same as above, the `map()` function does the same.

File: system/cache/providers/CFProvider.cfc (hunk 1 above)
Comment (sjdaniels) [reply system/cache/providers/CFProvider.cfc]: array().map() returns an array. This function is supposed to return a struct.

File: system/cache/providers/LuceeProvider.cfc (hunk 4 above)
Comment (sjdaniels) [reply system/cache/providers/LuceeProvider.cfc]: See above.

File: system/cache/providers/CFProvider.cfc (hunk 1 above)
Comment (lmajano) [reply system/cache/providers/CFProvider.cfc]: Ohhhhhhhhh, ok, then let’s switch it up to a `reduce()` function then, so we can eat our cake 🍰

Comment (sjdaniels) [reply system/cache/providers/CFProvider.cfc]: OK. All set.

**Pull request merged**